src/cpp/benders/benders_mpi/BendersMPI.cpp

```diff
@@ -293,6 +295,136 @@ SubProblemDataMap BendersMpi::get_subproblem_cut_package()
     return subproblem_data_map;
 }
 
+#if 0 
+std::vector<SubProblemDataMap> BendersMpi::split_subproblem_data_with_shuffle(std::vector<SubProblemDataMap>& gathered_subproblem_map, int n_cuts) 
+{
+    std::vector<std::pair<std::string, PlainData::SubProblemData>> allElements;
+    allElements.reserve(10000); 
+
+    for (auto& spMap : gathered_subproblem_map) {
+        for (auto it = spMap.begin(); it != spMap.end(); ) {
+            // Move key/value pair into the flat list
+            allElements.emplace_back(
+                it->first, 
+                std::move(it->second)
+            );
+            it = spMap.erase(it);
+        }
+    }
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::shuffle(allElements.begin(), allElements.end(), gen);
+
+    std::vector<SubProblemDataMap> result(n_cuts);
+
+    if (n_cuts <= 0 || allElements.empty())
+        return result;
+
+    size_t total = allElements.size();
+    size_t target_per_cut = (total + n_cuts - 1) / n_cuts; // ceiling division
+
+    size_t index = 0;
+    for (size_t cut = 0; cut < n_cuts && index < total; ++cut) {
+        size_t take = std::min(target_per_cut, total - index);
+
+        for (size_t k = 0; k < take; ++k) {
+            auto& elem = allElements[index++];
+            result[cut].emplace(
+                std::move(elem.first),
+                std::move(elem.second)
+            );
+        }
+    }
+
+    return result;
+}
+#endif 
+
+
+std::vector<std::vector<std::pair<std::string,int>>>  BendersMpi::split_subproblem_data_pairs(std::vector<SubProblemDataMap>& gathered_subproblem_map, int n_cuts)
+{
+    std::vector<std::vector<std::pair<std::string,int>>>  result(n_cuts) ; 
```

Comment:
Use a more explicit name
```suggestion
    AMoreExplicitStructName  subproblem_per_cut_ids(n_cuts) ; 
```